lib/model/query/submissions.js

```diff
@@ -79,24 +79,55 @@ const createVersion = (partial, deprecated, form, deviceIdIn = null, userAgentIn
   const deviceId = applyPipe(deviceIdIn, truncateString(255), blankStringToNull);
   const userAgent = applyPipe(userAgentIn, truncateString(255), blankStringToNull);
 
-  const _unjoiner = unjoiner(Submission, Submission.Def.into('currentVersion'));
-
   // we already do transactions but it just feels nice to have the cte do it all at once.
   return one(sql`
-with logical as (update submissions set "reviewState"='edited', "updatedAt"=clock_timestamp()
-  where id=${deprecated.submissionId} 
-  returning * )
-, upd as (update submission_defs set current=false where "submissionId"=${deprecated.submissionId} returning *)
-, def as (
-  ${_defInsert(deprecated.submissionId, partial, form.def.id, actorId, null, deviceId, userAgent)}
-)
-select ${_unjoiner.fields} from (
-  select logical.*, upd."userAgent" from logical
-  join upd on logical.id = upd."submissionId" and root
-) submissions
-join def submission_defs on submissions.id = submission_defs."submissionId"
-`)
-    .then(_unjoiner);
+    WITH
+      updatedSubmission AS (
+        UPDATE submissions
+          SET "reviewState"='edited'
+            , "updatedAt"=clock_timestamp()
+          WHERE id=${deprecated.submissionId}
+        RETURNING *
+      ),
+      deprecatedDef AS (
+        UPDATE submission_defs
+          SET current = FALSE
+          WHERE "submissionId"=${deprecated.submissionId}
+            AND current IS TRUE
+      ),
+      newDef AS (
+        INSERT INTO submission_defs ("submissionId", xml, "formDefId", "instanceId", "instanceName", "submitterId", "localKey", "encDataAttachmentName", "signature", "createdAt", root, current, "deviceId", "userAgent")
+          VALUES (${deprecated.submissionId}, ${sql.binary(partial.xml)}, ${form.def.id}, ${partial.instanceId}, ${partial.def.instanceName}, ${actorId}, ${partial.def.localKey}, ${partial.def.encDataAttachmentName}, ${partial.def.signature}, clock_timestamp(), null, true, ${deviceId}, ${userAgent})
+          RETURNING id AS "submissionDefId", "createdAt" as "submissionDefCreatedAt"
+      )
+    SELECT updatedSubmission.*
+         , submission_defs."userAgent"
+         , newDef.*
+      FROM updatedSubmission
+      JOIN submission_defs
+        ON "submissionId"=${deprecated.submissionId} AND root IS TRUE
+      CROSS JOIN newDef
+  `)
+    .then(({ submissionDefId, submissionDefCreatedAt, ...submissionData }) => // TODO/HACK: reassembling this from bits and bobs.
+      new Submission({ id: deprecated.submissionId, ...submissionData }, {
+        currentVersion: new Submission.Def({
```

Comment:
Can we disable these eslint rules for the block of code instead of each line?

```
/* eslint-disable key-spacing, no-multi-spaces */
....
....
/* eslint-enable key-spacing, no-multi-spaces */